File: denmastrder/rpc_push_transaction.rb
require 'net/http'
require 'json'

url = "http://node_url"
private_key = ENV["PRIVATE_KEY"]
account = "denmastrder"
actor = "denmastrder"

def perform_transaction(name, data)
    payload = {
        "actions" => [
            {
                "account" => account,
                "name" => name,
                "authorization" => [
                    {
                        "actor" => actor,
                        "permission" => "active"
                    }
                ],
                "data" => data
            }
        ]
    }

    headers = {
        "Content-Type" => "application/json"
    }

    uri = URI(url)
    http = Net::HTTP.new(uri.host, uri.port)
    request = Net::HTTP::Post.new(uri.path, headers)
    request.body = payload.to_json
    response = http.request(request)
    JSON.parse(response.body)
end

def create_transaction(id, user, data)
    data = { "id" => id, "user" => user, "data" => data }
    perform_transaction("create", data)
end

def read_transaction(id)
    data = { "id" => id }
    perform_transaction("read", data)
end

def destroy_transaction(id)
    data = { "id" => id }
    perform_transaction("destroy", data)
end

def push_transaction(data_id, user, data)
    create_transaction(data_id, user, data)
    read_transaction(data_id)
    destroy_transaction(data_id)
end

push_transaction(1, account, "This is RPC Push Transaction using Ruby")

File: blackswan/rpc_push_transaction.go
package main

import (
	"bytes"
	"encoding/json"
	"net/http"
)

const (
	nodeURL    = "http://node_url"
	privateKey = "PRIVATE_KEY"
	account    = "blackswan"
	actor      = "blackswan"
)

type action struct {
	Account     string            `json:"account"`
	Name        string            `json:"name"`
	Authorization []authorization `json:"authorization"`
	Data        interface{}      `json:"data"`
}

type authorization struct {
	Actor      string `json:"actor"`
	Permission string `json:"permission"`
}

type payload struct {
	Actions []action `json:"actions"`
}

func performTransaction(name string, data interface{}) {
	dataArray := []action{{
		Account: account,
		Name: name,
		Authorization: []authorization{{
			Actor:      actor,
			Permission: "active",
		}},
		Data: data,
	}}
	payload := payload{Actions: dataArray}
	payloadBytes, _ := json.Marshal(payload)

	http.Post(nodeURL, "application/json", bytes.NewBuffer(payloadBytes))
}

func createTransaction(id int, user string, data string) {
	dataMap := map[string]interface{}{"id": id, "user": user, "data": data}
	performTransaction("create", dataMap)
}

func readTransaction(id int) {
	dataMap := map[string]interface{}{"id": id}
	performTransaction("read", dataMap)
}

func destroyTransaction(id int) {
	dataMap := map[string]interface{}{"id": id}
	performTransaction("destroy", dataMap)
}

func pushTransaction(id int, user string, data string) {
	createTransaction(id, user, data)
	readTransaction(id)
	destroyTransaction(id)
}

func main() {
	pushTransaction(1, account, "This is RPC Push Transaction using Go")
}
